intake/container/__init__.py

```diff
@@ -20,3 +20,40 @@
 }
 
 __all__ = ['container_map']
+
+
+def upload(data, path, **kwargs):
+    """Given a concrete data object, store it at given location return Source
+
+    Use this function to publicly share data which you have created in your
+    python session. Intake will try each of the container types, to see if
+    one of them can handle the input data, and write the data to the path
+    given, in the format most appropriate for the data type, e.g., parquet for
+    pandas or dask data-frames.
+
+    With the DataSource instance you get back, you can add this to a catalog,
```

Comment:
Note that YAMLFileCatalog.add_source() will come later in this PR. I don't think it's meaningful for any other type of cat class (unless specifically developed for whatever service supplies the cat).